src/ui/public/visualize/visualize.js

```diff
@@ -99,18 +96,20 @@ define(function (require) {
           'transition-delay': loadingDelay
         };
 
-        // spy watchers
-        $scope.$watch('fullScreenSpy', applyClassNames);
-
-        $scope.$watchCollection('spy.mode', function (spyMode, oldSpyMode) {
+        function shouldHaveFullSpy() {
           var $visEl = getVisEl();
           if (!$visEl) return;
 
-          // if the spy has been opened, check chart height
-          if (spyMode && !oldSpyMode) {
-            $scope.fullScreenSpy = $visEl.height() < minVisChartHeight;
-          }
+          return ($visEl.height() < minVisChartHeight)
+            && _.get($scope.spy, 'mode.fill')
```

Comment:
Should these be `has` instead of `get`?  As it stands now, if either `mode.fill` or `mode.name` are not truthy (e.g. `0`), this will fail.  That may be the intended behavior, but I wanted to make sure.